controllers/common.go

```diff
@@ -775,17 +774,31 @@ func (r *MultiClusterHubReconciler) ensureNoClusterManagementAddOn(m *operatorv1
 	}
 
 	err = r.Client.Delete(context.TODO(), clusterMgmtAddon)
```

Comment:
Should we delete this with foreground? @mprahl

Reply:
https://kubernetes.io/docs/tasks/administer-cluster/use-cascading-deletion/

Reply:
Let's leave that to a separate PR since that requires more testing.

Reply:
@yiraeChristineKim that is the plan for this function. Can we add that update within this PR?